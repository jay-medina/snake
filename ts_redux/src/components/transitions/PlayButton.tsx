import React from 'react';

export interface PlayButtonProps {
  onClick(): void;
}

export const PlayButton: React.SFC<PlayButtonProps> = ({ onClick }) => (
  <div className="snake__start-screen-play-container">
    <button className="snake__start-screen-play" onClick={onClick}>
      Play
    </button>
  </div>
);

PlayButton.displayName = 'PlayButton';
