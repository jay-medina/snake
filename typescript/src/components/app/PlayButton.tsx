import React from 'react';

export const PlayButton: React.SFC<{ onClick: () => void }> = ({ onClick }) => {
  return (
    <div className="snake__start-screen-play-container">
      <button className="snake__start-screen-play" onClick={onClick}>
        Play
      </button>
    </div>
  );
};
