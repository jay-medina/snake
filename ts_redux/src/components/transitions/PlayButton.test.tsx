import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { PlayButton, PlayButtonProps } from './PlayButton';

describe('<PlayButton />', () => {
  let onClick: jest.Mock;
  let component: ShallowWrapper<PlayButtonProps>;

  beforeEach(() => {
    onClick = jest.fn();
    component = shallow(<PlayButton onClick={onClick} />);
  });

  it('renders the component', () => {
    expect(component).toMatchSnapshot();
  });

  describe('when clicked', () => {
    it('invokes the onClick callback', () => {
      component.find('button').simulate('click');
      expect(onClick).toHaveBeenCalled();
    });
  });
});
