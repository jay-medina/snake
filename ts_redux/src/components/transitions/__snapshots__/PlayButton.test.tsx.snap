// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayButton /> renders the component 1`] = `
<div
  className="snake__start-screen-play-container"
>
  <button
    className="snake__start-screen-play"
    onClick={[MockFunction]}
  >
    Play
  </button>
</div>
`;
